### PR TITLE
(feat) Patient name sorting

### PR DESCRIPTION
**Pull request by icrc-psousa**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary
When creating a group, no specific order for the patients is explicitly defined. Initially, the order established during the creation is maintained for the first session. However, for subsequent sessions, the order of patients changes, following the sequence in which they are retrieved from the database.

With this development the order of the patients within a group will always be alphabetical.

### CODE REVIEW

File: src/group-form-entry-workflow/attendance-table/AttendanceTable.tsx [hunk 1]
```diff
@@ -129,12 +130,16 @@ const AttendanceTable = ({ patients }) => {
...
+          {activeGroupMembers
+            .map((patientUuid, index) => {
+              return patients.find((patient) => patient.id === patientUuid);
+            })
+            .sort((a, b) => {
+              return getPatientName(a).localeCompare(getPatientName(b));
+            })
+            .map((patient, index) => {
+              return <PatientRow patient={patient} key={index} />;
+            })}
```
[the remark this answers is not in the document]
Comment (icrc-psousa) [reply src/group-form-entry-workflow/attendance-table/AttendanceTable.tsx]: Agreed, done!

File: src/add-group-modal/AddGroupModal.tsx [hunk 2]
```diff
@@ -59,11 +48,71 @@ const PatientRow = ({ patient, removePatient }) => {
...
+  const getPatients = async (uuids) => {
+    const results = await Promise.all(
+      uuids.map(async (uuid) => await fetchCurrentPatient(uuid))
+    );
+    setPatients(results);
+  };
+
+  useEffect(() => {
+    getPatients(patientList.map((obj) => obj.uuid));
+  }, [patientList]);
```
[the remark this answers is not in the document]
Comment (icrc-psousa) [reply src/add-group-modal/AddGroupModal.tsx]: Done! @denniskigen if possible have a look and confirm it's well implemented

**Comment by icrc-psousa** [comment]
> Thanks for working on this, @icrc-psousa, I've left some suggestions.

Thanks for your input @denniskigen , I've made the suggested changes. I've also fixed the sorting on the modal not to consider the patient id so that the patient list order is always the same.

File: src/add-group-modal/AddGroupModal.tsx [hunk 3]
```diff
@@ -59,11 +48,80 @@ const PatientRow = ({ patient, removePatient }) => {
...
+  const { patientList, removePatient } = props;
+
+  const [patients, setPatients] = useState([]);
+
+  useEffect(() => {
+    let ignore = false;
+
+    const getPatients = async (uuids) => {
+      return await Promise.all(uuids.map((uuid) => fetchCurrentPatient(uuid)));
+    };
```
Comment (ibacher) [inline src/add-group-modal/AddGroupModal.tsx:66]: Note that the `async` and `await` here are superfluous.

**Review by ibacher — APPROVED** [review]

**Review by icrc-jofrancisco — APPROVED** [review]

**Pull request merged**